doc/conditional_independence.rst

```diff
@@ -0,0 +1,180 @@
+  on the permutated datasets to estimate a pvalue.
+
+  It is worth noting that if one has "robust" estimates of k-NN using for example a model that
+  is effective in high-dimensions, then the KSG estimator for CMI may be quite good. For example,
+  one can use variants of Random Forests to generate adaptive nearest-neighbor estimates in high-dimensions
+  or on manifolds, such that the KSG estimator is still quite good.
+
+.. autosummary::
+   :toctree: generated/
+
+    CMITest
+
+- The Classifier Divergence approach estimates CMI using a classification model.
+
+.. autosummary::
+   :toctree: generated/
+
+    ClassifierCMITest
+
+- Direct posterior estimates can be implemented with a classification model by directly
+  estimating :math:`P(y|x)` and :math:`P(y|x,z)`, which can be used as plug-in estimates
+  to the equation for CMI.
+
+Partial (Pearson) Correlation
+-----------------------------
+Partial correlation based on the Pearson correlation is equivalent to CMI in the setting
+of normally distributed data. Computing partial correlation is fast and efficient and
+thus attractive to use. However, this **relies on the assumption of multivariate Gaussianity**,
+which may be unrealistic in certain datasets.
+
+.. autosummary::
+   :toctree: generated/
+
+    FisherZCITest
+
+Discrete, Categorical and Binary Data
+-------------------------------------
+If one has discrete data, then the test to use is based on Chi-square tests. The :math:`G^2`
+class of tests will construct a contingency table based on the number of levels across
+each discrete variable. An exponential amount of data is needed for increasing levels
+for a discrete variable.
+
+.. autosummary::
+   :toctree: generated/
+
+    GSquareCITest
+
+Kernel-Approaches
+-----------------
+Kernel-based tests are attractive since they are semi-parametric and use kernel-based ideas
+that have been shown to be robust in the machine-learning field. The Kernel CI test is a test
+that computes a test statistic from kernels of the data and uses permutation testing to
+generate samples from the null distribution :footcite:`Zhang2011`, which are then used to
+estimate a pvalue.
+
+.. autosummary::
+   :toctree: generated/
+
+    KernelCITest
+
+Classifier-based Approaches
+---------------------------
+Another suite of approaches that rely on permutation testing is the classifier-based approach.
+By shuffling data in an intelligent way, one can setup a hypothesis test for CI based on the
+predicted probabilities from a classification-model. Intuitively, if the shuffled data is similar 
+to the unshuffled data, such that the classification-model achieves non-trivial performance
+(e.g. >50\% accuracy on a balanced dataset), then one fails to reject the null hypothesis and would
+state that the original data was in fact CI :footcite:`Sen2017model`.
+
+.. autosummary::
+   :toctree: generated/
+
+    ClassifierCITest
```

Comment:
Not relevant for this PR: But why isn't there one based on regression? So, imply checking whether the model improves when you add the other variable(s)?

Reply:
I just never implemented one because I saw dowhy had it. We can consolidate now that we moving CI stuff to pywhy-stats :). I think upon skimming, the meta-method for helping generate the pvalue can be consolidated with a suite of meta-methods for pvalue estimation.